compiler/rustc_middle/src/ty/context.rs

```diff
@@ -362,6 +362,14 @@ pub struct GeneratorInteriorTypeCause<'tcx> {
     pub expr: Option<hir::HirId>,
 }
 
+#[derive(TyEncodable, TyDecodable, Clone, Debug, HashStable)]
+pub struct GeneratorDiagnosticData<'tcx> {
```

Comment:
Could you leave a documentation comment describing what this struct is for, etc.